### PR TITLE
Make singleton instances on plugins only accessible through getters

### LINKED ISSUE #328: Get instances of singletons via functions instead of global variables
Instead of exposing variables like `Instance` on a plugin, consider exposing a function which returns a singleton instance. For example, instead of having `drngplugin.Instance`, change it to `drngplugin.DRNG()`/`drngplugin.Drng()`. Those functions can also use `sync.Once` to init the given singleton instance.

### PR DESCRIPTION
**Pull request by acha-bill**
Encapsulation
#328

### CODE REVIEW

File: plugins/webapi/plugin.go [hunk 1]
```diff
@@ -18,25 +19,44 @@ import (
...
+	plugin     *node.Plugin
+	pluginOnce sync.Once
+	// server is the web API server.
+	server *echo.Echo
+	serverOnce sync.Once
 
 	log *logger.Logger
 )
 
+// Plugin gets the plugin instance
```
Comment (luca-moser) [inline plugins/webapi/plugin.go:32]:
```suggestion
// Plugin gets the plugin instance.
```

File: plugins/webapi/plugin.go [hunk 2]
```diff
@@ -18,25 +19,44 @@ import (
...
+func Plugin() *node.Plugin {
+	pluginOnce.Do(func() {
+		plugin = node.NewPlugin(PluginName, node.Enabled, configure, run)
+	})
+	return plugin
+}
+
+// Server gets the server instance
```
Comment (luca-moser) [inline plugins/webapi/plugin.go:40]:
```suggestion
// Server gets the server instance.
```

File: plugins/webapi/spammer/plugin.go [hunk 3]
```diff
@@ -8,22 +8,33 @@ import (
...
-var Plugin = node.NewPlugin(PluginName, node.Disabled, configure, run)
+var (
+	// plugin is the plugin instance of the spammer plugin.
+	plugin *node.Plugin
+	once sync.Once
+	log *logger.Logger
+)
 
-var log *logger.Logger
+// Plugin gets the plugin instance
```
Comment (luca-moser) [inline plugins/webapi/spammer/plugin.go:26]:
```suggestion
// Plugin gets the plugin instance.
```

File: plugins/webapi/value/plugin.go [hunk 4]
```diff
@@ -8,20 +8,30 @@ import (
...
 
 var (
-	// Plugin is the plugin instance of the web API DRNG endpoint plugin.
-	Plugin = node.NewPlugin(PluginName, node.Enabled, configure)
+	// plugin is the plugin instance of the web API DRNG endpoint plugin.
+	plugin *node.Plugin
+	once sync.Once
 )
 
+// Plugin gets the plugin instance
```
Comment (luca-moser) [inline plugins/webapi/value/plugin.go:23]:
```suggestion
// Plugin gets the plugin instance.
```

File: plugins/webauth/webauth.go [hunk 5]
```diff
@@ -18,20 +19,29 @@ import (
...
-	// Plugin is the plugin instance of the web API auth plugin.
-	Plugin     = node.NewPlugin(PluginName, node.Disabled, configure)
+	// plugin is the plugin instance of the web API auth plugin.
+	plugin     *node.Plugin
+	once 	   sync.Once
 	log        *logger.Logger
 	privateKey string
 )
 
+// Plugin gets the plugin instance
```
Comment (luca-moser) [inline plugins/webauth/webauth.go:29]:
```suggestion
// Plugin gets the plugin instance.
```

File: dapps/valuetransfers/dapp.go [hunk 6]
```diff
@@ -67,15 +69,41 @@ var (
...
 
+// App gets the plugin instance
+func App() *node.Plugin {
+	appOnce.Do(func () {
+		app = node.NewPlugin(PluginName, node.Enabled, configure, run)
+	})
+	return app
+}
+
+// Tangle gets the tangle instance
```
Comment (lzpap) [inline dapps/valuetransfers/dapp.go]: add here the original comment from line 52. Would be nice to have a descriptive comment for all exported functions, so my editor tells me exactly what a `valuetransfers.Tangle()` is when I want to use it from outside. Same goes for all the newly introduced singleton getters  in this PR below.

**Comment by luca-moser** [comment]
Please fix the conflicts so we can merge it

**Pull request merged**